wgpu/src/backend/web.rs

```diff
@@ -2,15 +2,42 @@
 
 use js_sys::Promise;
 use std::{
+    any::Any,
     cell::RefCell,
     fmt,
     future::Future,
+    num::NonZeroU128,
     ops::Range,
     pin::Pin,
     rc::Rc,
     task::{self, Poll},
 };
-use wasm_bindgen::{prelude::*, JsCast};
+use wasm_bindgen::{
+    convert::{FromWasmAbi, IntoWasmAbi},
+    prelude::*,
+    JsCast,
+};
+
+use crate::{
+    context::{ObjectId, QueueWriteBuffer},
+    UncapturedErrorHandler,
+};
+
+#[cfg(feature = "expose-ids")]
+static NEXT_ID: std::sync::atomic::AtomicU64 = std::sync::atomic::AtomicU64::new(0);
```

Comment:
To clean up this code, we can use cfg_if and put the static inside the function.